src/inmanta/agent/agent_new.py

```diff
@@ -181,48 +181,70 @@ async def start_working(self) -> None:
         await self.scheduler.start()
         self._enable_time_triggers()
 
-    async def stop_working(self) -> None:
+    async def stop_working(self, timeout: Optional[int] = None) -> None:
         """Stop working, connection lost"""
         if not self.working:
             return
         self.working = False
         self._disable_time_triggers()
         await self.executor_manager.stop()
+        if timeout is not None:
+            await self.executor_manager.join([], timeout=timeout)
         await self.scheduler.stop()
 
     @protocol.handle(methods_v2.update_agent_map)
     async def update_agent_map(self, agent_map: dict[str, str]) -> None:
         # Not used here
         pass
 
-    async def unpause(self, name: str) -> Apireturn:
-        if name != AGENT_SCHEDULER_ID:
-            return 404, "No such agent"
+    @protocol.handle(methods.set_state)
+    async def set_state(self, agent: str, enabled: bool) -> Apireturn:
+        # if agent == AGENT_SCHEDULER_ID:
+        #     return (
+        #         400,
+        #         "Invalid request, cannot pause the scheduler! The closest action to that would be to pause the environment",
+        #     )
+
+        if agent == AGENT_SCHEDULER_ID:
```

Comment:
Need to be uncommented and removed

Reply:
But in order to do so, I need help with this comment: https://github.com/inmanta/inmanta-core/pull/8141#discussion_r1803063778